core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -253,51 +333,60 @@ public String validateChallenge(String challengeXdr)
 
       // increment counter
       sep10ChallengeValidatedCounter.increment();
```

Comment:
Should the counter be incremented here?

Reply:
This is actually a bug. We should returned inside the try block and the end of the function.